src/ctapipe/irf/preprocessing.py

```diff
@@ -206,93 +563,240 @@ class EventLoader(Component):
+                self.log.error("No observation information available in the DL2 file")
+                raise ValueError("Missing observation information in the DL2 file")
+            elif len(obs_all_info_table) != len(obs_info_table):
+                self.log.error(
+                    "Scheduling blocks are missing for some observation block"
+                )
+                raise ValueError(
+                    "Scheduling blocks are missing for some observation block"
                 )
 
-        sim_info = SimulatedEventsInfo(
-            n_showers=show["n_entries"].sum(),
-            energy_min=sim["energy_range_min"].quantity[0],
-            energy_max=sim["energy_range_max"].quantity[0],
-            max_impact=sim["max_scatter_range"].quantity[0],
-            spectral_index=sim["spectral_index"][0],
-            viewcone_max=sim["max_viewcone_radius"].quantity[0],
-            viewcone_min=sim["min_viewcone_radius"].quantity[0],
-        )
+            # Check for obs ids
+            if len(np.unique(obs_all_info_table["obs_id"])):
+                self.log.warning(
+                    "Multiple observations included in the DL2 file, only the id of the first observation will be reported in the DL3 file, data from all observations will be included"
+                )
+            meta["obs_id"] = np.unique(obs_all_info_table["obs_id"])[0]
+
+            # Extract GTI
+            list_gti = []
+            mask_nan = np.isnan(obs_all_info_table["actual_duration"])
+            if np.sum(mask_nan) > 0:
+                self.log.warning("Duration of the run is nan, replaced with zero")
+                obs_all_info_table["actual_duration"][mask_nan] = 0.0 * u.s
+            for i in range(len(obs_all_info_table)):
+                start_time = Time(obs_all_info_table["actual_start_time"][i]).tai
+                stop_time = start_time + TimeDelta(
+                    obs_all_info_table["actual_duration"][i]
+                    * obs_all_info_table["actual_duration"].unit
+                )
+                list_gti.append((start_time, stop_time))
+            meta["gti"] = list_gti
 
-        return sim_info, PowerLaw.from_simulation(sim_info, obstime=obs_time)
+            # Dead time fraction
+            self.log.warning(
+                "Dead time fraction is not read from the file, a default value of 1 is used"
+            )
+            meta["dead_time_fraction"] = 1.0
+
+            # Extract pointing information
+            if len(np.unique(obs_all_info_table["pointing_mode"])) != 1:
+                self.log.error("Multiple pointing mode are not supported")
+                raise ValueError("Multiple pointing mode are not supported")
+            meta["pointing"] = {}
+            meta["pointing"]["pointing_mode"] = PointingMode(
+                obs_all_info_table["pointing_mode"][0]
+            ).name
+            list_pointing = []
+            for i in range(len(obs_all_info_table)):
```

Comment:
You can iterate over the rows of the table and if you convert it to `QTable`, the items will be already Quantity instances, so no need of the unit handling you do below.